llmfoundry/models/hf/hf_fsdp.py

```diff
@@ -141,15 +142,20 @@ def prepare_hf_causal_lm_model_for_fsdp(model: Union[PreTrainedModel,
     """
     causal_base_model = hf_get_causal_base_model(model)
 
-    # OPT has an extra layer of wrapping, so special case here
-    if isinstance(causal_base_model, OPTDecoder):
+    # OPT and olmo have an extra layer of wrapping, so special case here
+    if isinstance(causal_base_model,
+                  OPTDecoder) or model.config.model_type == 'olmo':
         underlying_model = maybe_get_underlying_model(model)
         underlying_model.model._fsdp_wrap = False
     model_block = hf_get_hidden_layers(causal_base_model)
     lm_head = model.get_output_embeddings()
-    # some models (OPT) implement .get_input_embeddings for the causal subclass
-    # but all of them implement it for the base model
-    tied_embeddings = causal_base_model.get_input_embeddings()
+    # Try to get input embeddings from the transformer backbone
+    # and then from the XXXForCausalLM
+    try:
+        tied_embeddings = causal_base_model.get_input_embeddings()
+    except:
+        tied_embeddings = model.get_input_embeddings()
```

Comment:
why is this necessary?

Reply:
cause models have this defined in different places